dll/shellext/netshell/lanconnectui.cpp

```diff
@@ -26,15 +26,16 @@ CNetConnectionPropertyUi::~CNetConnectionPropertyUi()
 }
 
 VOID
-AddItemToListView(HWND hDlgCtrl, PNET_ITEM pItem, LPWSTR szName, BOOL bChecked)
+AddItemToListView(HWND hDlgCtrl, PNET_ITEM pItem, LPWSTR szName, BOOL bChecked, UINT Image)
 {
     LVITEMW lvItem;
 
-    ZeroMemory(&lvItem, sizeof(lvItem));
-    lvItem.mask  = LVIF_TEXT | LVIF_PARAM;
+    lvItem.mask  = LVIF_TEXT | LVIF_PARAM | LVIF_IMAGE;
     lvItem.pszText = szName;
     lvItem.lParam = (LPARAM)pItem;
-    lvItem.iItem = ListView_GetItemCount(hDlgCtrl);
+    lvItem.iImage = Image;
+    lvItem.iItem = 0x7fff;
```

Comment:
Is this the fix for the checkbox, or an optimization?

Reply:
optimization.

The bug was swapped values for stateMask and state .

Reply:
i'm not sure why 0x7fff has to be used here, and not any other value.

Reply:
> i'm not sure why 0x7fff has to be used here, and not any other value

This is the largest supported value on Windows 95 IIRC.

Reply:
> This is the largest supported value on Windows 95 IIRC. This is what really old school sample code uses.

But we code for NT type of OSes, that's why I'm not very fond of that optimization.
What if you use `-1` as the value? Would it work when testing on Windows (XP/2k3/Vista+)?

Reply:
> What if you use `-1` as the value?

No, the value cannot be negative. 0x7fffffff would probably work.